### PR TITLE
Simplify mock tests

### PR DESCRIPTION
**Pull request by chrieke**
- Uses requests_mock fixture instead of Mocker with context menu.
- Puts nearly all mock http adresses inside the fixture. Makes the actual tests much more readable & saves time to create new tests. A few url mocks need to remain in the specific test, e.g. when the `.info` mock when creating a new object.
- Uses fixed variables for workflow_name, job_name etc. to be more consistent.

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -71,24 +231,84 @@ def project_live(auth_live):
...
+    )
+    json_workflow_info = {
+        "data": {
+            "name": WORKFLOW_NAME,
+            "id": WORKFLOW_ID,
+            "xyz": 789,
+        },
+        "error": {},
+    }
+    requests_mock.get(url=url_workflow_info, json=json_workflow_info)
```
Comment (markusuwe) [inline tests/fixtures.py:249]: Why is this preferable to using a context manager?

**Review by markusuwe — APPROVED** [review]
LGTM
Can't pretend I looked over this huge list of changes in detail.

**Review by rodrigoalmeida94 — APPROVED** [review]
LGTM! One suggestiom to make this even more flexible, in light of what we discussed yeasterday re mock objects.

**Review by nekhtiari — APPROVED** [review]

File: tests/fixtures.py (hunk 1 above)
Comment (chrieke) [reply tests/fixtures.py:249]: With context manager the function call needs to be inside the context, wouldn't allow to put it in the fixture.

File: tests/fixtures.py [hunk 2]
```diff
@@ -14,34 +14,142 @@
...
+
+JOB_ID = "job_id_123"
+JOB_ID_2 = "jobid_456"
+JOB_NAME = "job_name_123"
+
+JOBTASK_ID = "jobtask_id_123"
+JOBTASK_NAME = "jobtask_name_123"
+
+
+json_workflow_tasks = {
```
[the remark this answers is not in the document]
Comment (chrieke) [reply tests/fixtures.py:38]: Using caps now, thanks

File: tests/fixtures.py [hunk 3]
```diff
@@ -14,34 +14,142 @@
...
+    url_get_token = (
+        f"https://{auth.project_id}:{auth.project_api_key}@api.up42."
+        f"{auth.env}/oauth/token"
+    )
+    json_get_token = {"data": {"accessToken": TOKEN}}
+    requests_mock.post(
+        url=url_get_token,
+        json=json_get_token,
+    )
+
```
[the remark this answers is not in the document]
Comment (chrieke) [reply tests/fixtures.py:125]: Nice! I created a ticket UP-4935 to pick this up. Keeping the traditional fixtures for now, will have more visibility if we do it separatly and add it with an example in the docs.

**Pull request merged**